### PR TITLE
Fix `MacosPulldownMenuItem.onTap` doesn't open alert dialog

### LINKED ISSUE #516: MacosPulldownMenuItem onTap doesnt call showMacosAlertDialog
**Description**

When i call  `showMacosAlertDialog` from `ToolBarIconButton` `onPressed` works as expected, i see the dialog poping up

When i call  `showMacosAlertDialog` from `MacosPulldownMenuItem` `onTap` does not call the function

**Steps To Reproduce**

Run the sample app below

<details>
<summary>Code sample</summary>

```
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:macos_ui/macos_ui.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MacosApp(
      title: 'macos_ui example',
      theme: MacosThemeData.light(),
      darkTheme: MacosThemeData.dark(),
      themeMode: ThemeMode.system,
    );
  }
}

class Demo extends StatefulWidget {
  const Demo({super.key});

  @override
  _DemoState createState() => _DemoState();
}

class _DemoState extends State<Demo> {
  double ratingValue = 0;
  double sliderValue = 0;
  bool value = false;

  int pageIndex = 0;

  @override
  Widget build(BuildContext context) {
    return const MacosWindow(
      child: AppsScreen(),
    );
  }
}

class AppsScreen extends StatelessWidget {
  const AppsScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return MacosScaffold(
      toolBar: ToolBar(
        title: Text('Apps'),
        actions: <ToolbarItem>[
          ToolBarIconButton(
            label: 'Release for all',
            icon: const MacosIcon(
              CupertinoIcons.wand_stars,
            ),
            showLabel: true,
            onPressed: () => showMacosAlertDialog(
              context: context,
              builder: (BuildContext context) {
                return MacosAlertDialog(
                  title: const Text('Release for all'),
                  message: const Text('Are you sure you want to release this app for all?'),
                  primaryButton: PushButton(
                    onPressed: () => Navigator.of(context).pop(),
                    controlSize: ControlSize.large,
                    child: const Text('Cancel'),
                  ),
                  secondaryButton: PushButton(
                    controlSize: ControlSize.large,
                    onPressed: () => Navigator.of(context).pop(),
                    child: const Text('Release'),
                  ),
                  appIcon: const FlutterLogo(size: 56),
                );
              },
            ),
          ),
          ToolBarPullDownButton(
              label: 'Actions',
              tooltipMessage: 'actions about apps',
              icon: CupertinoIcons.gear_alt,
              items: <MacosPulldownMenuEntry>[
                MacosPulldownMenuItem(
                  label: 'Reset IAP cache',
                  title: Text('Reset IAP cache'),
                  onTap: ()   =>  showMacosAlertDialog(
                    context: context,
                    builder: (BuildContext context) {
                      return MacosAlertDialog(
                        title: const Text('Release for all'),
                        message: const Text('Are you sure you want to release this app for all?'),
                        primaryButton: PushButton(
                          onPressed: () => Navigator.of(context).pop(),
                          controlSize: ControlSize.large,
                          child: const Text('Cancel'),
                        ),
                        secondaryButton: PushButton(
                          controlSize: ControlSize.large,
                          onPressed: () => Navigator.of(context).pop(),
                          child: const Text('Release'),
                        ),
                        appIcon: const FlutterLogo(size: 56),
                      );
                    },
                  ),
                ),

              ]),
        ],
      ),
      children: <Widget>[
        ContentArea(builder: (BuildContext context, ScrollController scrollController) {
          return ListView.builder(
            controller: scrollController,
            itemCount: 10,
            itemBuilder: (BuildContext context, int index) {
              return MacosListTile(
                title: Text('App $index'),
                subtitle: Text('Subtitle $index'),
              );
            },
          );
        }),
      ],
    );
  }
}

```

</details>

**Expected behavior**

`MacosPulldownMenuItem` `onTap` should call the `showMacosAlertDialog`

**Screenshots**

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes [MacosPulldownMenuItem onTap doesnt call showMacosAlertDialog ](https://github.com/macosui/macos_ui/issues/516)

### Code sample

<details>
<summary>expand to view the code sample</summary> 

```dart
import 'package:flutter/cupertino.dart';
import 'package:macos_ui/macos_ui.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MacosApp(
      debugShowCheckedModeBanner: false,
      home: HomePage(),
    );
  }
}

class HomePage extends StatelessWidget {
  const HomePage({super.key});

  @override
  Widget build(BuildContext context) {
    return MacosScaffold(
      toolBar: ToolBar(
        title: const Text('macos_ui'),
        actions: [
          ToolBarPullDownButton(
            label: 'Actions',
            icon: CupertinoIcons.ellipsis_circle,
            tooltipMessage: 'Perform tasks with the selected items',
            items: [
              MacosPulldownMenuItem(
                label: 'New Folder',
                title: const Text('New Folder'),
                onTap: () => showMacosAlertDialog(
                  context: context,
                  builder: (context) => MacosAlertDialog(
                    appIcon: const FlutterLogo(size: 64),
                    title: const Text('Title'),
                    message: const Text('Message'),
                    //horizontalActions: false,
                    primaryButton: PushButton(
                      controlSize: ControlSize.large,
                      onPressed: Navigator.of(context).pop,
                      child: const Text('Label'),
                    ),
                  ),
                ),
              ),
              MacosPulldownMenuItem(
                label: 'Open',
                title: const Text('Open'),
                onTap: () => debugPrint('Opening...'),
              ),
            ],
          ),
        ],
      ),
    );
  }
}
```

</details>

### Preview 

Screenshot 2024-10-04 at 00 40 39

## Pre-launch Checklist

- [x] I have incremented the package version as appropriate and updated `CHANGELOG.md` with my changes 
- [ ] I have added/updated relevant documentation 
- [x] I have run "optimize/organize imports" on all changed files
- [x] I have addressed all analyzer warnings as best I could

### CODE REVIEW

File: lib/src/buttons/pulldown_button.dart [hunk 1]
```diff
@@ -65,8 +65,8 @@ class _MacosPulldownMenuItemButtonState
     final MacosPulldownMenuEntry menuEntity =
         widget.route.items[widget.itemIndex].item!;
     if (menuEntity is MacosPulldownMenuItem) {
-      menuEntity.onTap?.call();
       Navigator.pop(context);
```
Comment (GroovinChip) [inline lib/src/buttons/pulldown_button.dart]: Can we make this `.of(context).pop()` please?

**Review by GroovinChip — CHANGES_REQUESTED** [review]
Thanks for the PR @TahaTesser !!! Just one small change request and then we are good to go

**Review by GroovinChip — APPROVED** [review]
LGTM! 🚀

**Comment by GroovinChip** [comment]
Just gonna run the checks manually since you're a first time contributor

**Comment by GroovinChip** [comment]
Looks like we've got some failing tests

**Comment by TahaTesser** [comment]
>  Looks like we've got some failing tests

For some reason test passes locally with FlutterLogo but not in the CI so i just replaced it.

**Pull request merged**